app/src/main/java/com/example/tasky/util/Screen.kt

```diff
@@ -0,0 +1,11 @@
+package com.example.tasky.util
+
+import com.example.tasky.common.Constants.ROUTE_LOGIN
+import com.example.tasky.common.Constants.ROUTE_REGISTER
+import com.example.tasky.common.Constants.ROUTE_SPLASH
+
+sealed class Screen(val route: String) {
+    object SplashScreen: Screen(ROUTE_SPLASH)
```

Comment:
I highly recommend to not create a separate splash screen. Android already has its own splash screen and that can lead to 2 splash screens that will be shown. Instead, you can use the official splash screen API which also lets you show the splash screen while checking if you're still logged in or not.

I explain it in this video: https://www.youtube.com/watch?v=Loo4i5IrZ4Y